wgengine/pendopen.go

```diff
@@ -30,6 +30,9 @@ func debugConnectFailures() bool {
 
 type pendingOpenFlow struct {
 	timer *time.Timer // until giving up on the flow
+
+	// guaded by userspaceEngine.mu:
+	problem *packet.TailscaleRejectReason // non-nil if we got a non-terminal TSMP reject
```

Comment:
rather than a pointer, maybe use a zero value of the enum? \0 is an easy value for that, and means comparisons look a bit more distinctive? No strong feeling.

Reply:
Done.